scipy/stats/_continuous_distns.py

```diff
@@ -661,6 +662,66 @@ def _munp(self, n, c, d):
 burr = burr_gen(a=0.0, name='burr')
 
 
+class burr12_gen(rv_continuous):
+    """A Burr (Type XII) continuous random variable.
+
+    %(before_notes)s
+
+    See Also
+    --------
+    fisk : a special case of either `burr` or ``burr12`` with ``d = 1``
+    burr : Burr Type III distribution
+
+    Notes
+    -----
+    The probability density function for `burr` is::
+
+        burr12.pdf(x, c, d) = c * d * x**(c-1) * (1+x**(c))**(-d-1)
+
+    for ``x > 0``.
+
+    `burr12` takes ``c`` and ``d`` as shape parameters.
+
+    This is the PDF corresponding to the twelfth CDF given in Burr's list;
+    specifically, it is equation (20) in Burr's paper [1]_.
+
+    %(after_notes)s 
+    
+    The Burr type 12 distribution is also sometimes referred to as
+    the Singh-Maddala distribution from NIST.
+
+    References
+    ----------
+    .. [1] Burr, I. W. "Cumulative frequency functions", Annals of
+       Mathematical Statistics, 13(2), pp 215-232 (1942).
+
+    %(example)s
+
+    """
+    def _pdf(self, x, c, d):
+        return np.exp(self._logpdf(x, c, d))
+        
+    def _logpdf(self, x, c, d):
+        return log(c) + log(d) + special.xlogy(c-1, x) + special.xlog1py(-d-1, x**c)
+
+    def _cdf(self, x, c, d):
+        return 1-self._sf(x, c, d)
```

Comment:
Please contain the pep8 fixes to the parts which you actually touch for other reasons. Do not go haywire correcting all pep8 violations in that file or elsewhere: this makes diffs larger than necessary.